pkg/provisioner/resourcehandlers.go

```diff
@@ -283,28 +283,19 @@ func getExistingConfigMap(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interfac
 	return configMap, err
 }
 
-func deleteSecretAndWait(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) error {
-	name := composeSecretName(obc)
-	logD.Info("deleting secret", "secret name", name)
+func deleteSecretAndWait(sec *corev1.Secret, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) error {
+	logD.Info("deleting secret", "secret name", sec.Name)
 
-	secret, err := c.CoreV1().Secrets(obc.Namespace).Get(context.TODO(), name, metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Already deleted
-			return nil
-		}
-		return fmt.Errorf("failed to get secret in order to delete it: %v", err)
-	}
-	err = releaseSecret(secret, c)
+	err := releaseSecret(sec, c)
 	if err != nil {
 		return fmt.Errorf("failed to release secret for deletion: %v", err)
 	}
-	err = c.CoreV1().Secrets(obc.Namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
+	err = c.CoreV1().Secrets(sec.Namespace).Delete(context.TODO(), sec.Name, metav1.DeleteOptions{})
 	if err != nil {
 		return err
 	}
 	err = wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
```

Comment:
why do we need to poll/GET the secret and CM? If the `Delete` works then we know the timestamp is set and we've already removed the finalize, so it's just a matter of (small) time and the secret will be deleted. I can't recall why we should poll here...

Reply:
This function `deleteSecretAndWait` which waits for the secret to be deleted before returning. We need to wait for the secret to be gone before creating the new secret to replace the one being deleted.